pkg/cluster/compute/affinityrule/affinityrule.go

```diff
@@ -51,16 +70,35 @@ func CreateAffinityRule(whost *host.WmiHost, name string, ruleType int) (affinit
 	// added layer of protection, GetClusterAffinityRule can return not found immediately after creating the affinity rule
 	// if this happens, we will retry the get operation a few times before returning an error
 	maxAttempts := 5
-	for i := 0; i < maxAttempts; i++ {
-		affinityRule, err := GetAffinityRule(whost, name)
+	for i := 1; i <= maxAttempts; i++ {
+		affinityRule, err = GetAffinityRule(whost, name)
 		if err != nil {
-			if errors.IsNotFound(err) {
+			if errors.IsNotFound(err) && i < maxAttempts {
 				time.Sleep(2 * time.Second)
 				continue
 			}
+			return
+		}
+		break
+	}
+	defer func() {
+		if err != nil {
+			affinityRule.RemoveAffinityRule()
+			affinityRule.Close()
+		}
+	}()
+
+	isAntiAffinityRule := (ruleType == int(DifferentFaultDomain) || ruleType == int(DifferentNode))
+	if isAntiAffinityRule && !strict {
+		setSoftAntiAffinity, err := isSoftAntiAffinitySupported(whost)
+		if err != nil {
 			return nil, err
 		}
-		return affinityRule, nil
+		if setSoftAntiAffinity {
+			if _, err := affinityRule.InvokeMethod("SetAffinityRule", int(ruleType), 1 /* Enabled */, 1 /* SoftAntiAffinity */); err != nil {
+				return nil, err
+			}
+		}
```

Comment:
Either return an error or atleast trace this condition

Reply:
Returning error if soft anti affinity is not supported